File: blurdetect.m
%BlurDetector - This code is used to detect if a given image is blurry or
%of good quality
%Author - Ritesh Chidambaram
%Email-riteshchidambaram@gmail.com
im1=imread('filename');%give your filename
gray1=rgb2gray(im1);%converts to grayscale
sizfilt=round(size(gray1)/20);%takes into consideration the size of the image while filtering
imm=medfilt2(gray1,sizfilt);%median filter
diffim=imm-gray1;%Calculates the difference image
% var1=var(im2double(gray1(:)));
% var2=var(im2double(imm(:)));
diffvar=var(im2double(diffim(:)));%VAriance of difference image
if diffvar > 0.004%thresholding
    disp("Not blurry");
else
    disp("Blurry");
end

File: blurDetect.py
import cv2
import sys
import numpy as np
import math
file_location=sys.argv[1];
im1=cv2.imread(file_location,0)
diffvar=cv2.Laplacian(im1, cv2.CV_64F).var()
if diffvar>100:
	print("Not Blurry")
else:
	print("Blurry");
print(diffvar)


